pkg/cmd/secret/set/set.go

```diff
@@ -0,0 +1,216 @@
+						"--visibility not supported for repository secrets; did you mean to pass --org?")}
+				}
+
+				if opts.Visibility != shared.VisAll && opts.Visibility != shared.VisPrivate && opts.Visibility != shared.VisSelected {
+					return &cmdutil.FlagError{Err: errors.New(
+						"--visibility must be one of `all`, `private`, or `selected`")}
+				}
+			}
+
+			if cmd.Flags().Changed("repos") && opts.Visibility != shared.VisSelected {
+				return &cmdutil.FlagError{Err: errors.New(
+					"--repos only supported when --visibility='selected'")}
+			}
+
+			if opts.Visibility == shared.VisSelected && len(opts.RepositoryNames) == 0 {
+				return &cmdutil.FlagError{Err: errors.New(
+					"--repos flag required when --visibility='selected'")}
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return setRun(opts)
+		},
+	}
+	cmd.Flags().StringVar(&opts.OrgName, "org", "", "List secrets for an organization")
+	cmd.Flags().Lookup("org").NoOptDefVal = "@owner"
+	cmd.Flags().StringVarP(&opts.Visibility, "visibility", "v", "private", "Set visibility for an organization secret: `all`, `private`, or `selected`")
+	cmd.Flags().StringSliceVarP(&opts.RepositoryNames, "repos", "r", []string{}, "List of repository names for `selected` visibility")
+	cmd.Flags().StringVarP(&opts.Body, "body", "b", "-", "Provide either a literal string or a file path; prepend file paths with an @. Reads from STDIN if not provided.")
+
+	return cmd
+}
+
+func setRun(opts *SetOptions) error {
+	body, err := getBody(opts)
+	if err != nil {
+		return fmt.Errorf("did not understand secret body: %w", err)
+	}
+
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("could not create http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+
+	var baseRepo ghrepo.Interface
+	if opts.OrgName == "" || opts.OrgName == "@owner" {
+		baseRepo, err = opts.BaseRepo()
+		if err != nil {
+			return fmt.Errorf("could not determine base repo: %w", err)
+		}
+	}
+
+	host := ghinstance.OverridableDefault()
+	if opts.OrgName == "@owner" {
+		opts.OrgName = baseRepo.RepoOwner()
+		host = baseRepo.RepoHost()
+	}
+
+	var pk *PubKey
+	if opts.OrgName != "" {
+		pk, err = getOrgPublicKey(client, host, opts.OrgName)
+	} else {
+		pk, err = getRepoPubKey(client, baseRepo)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to fetch public key: %w", err)
+	}
+
+	eBody, err := box.SealAnonymous(nil, body, &pk.Raw, opts.RandomOverride)
+	if err != nil {
+		return fmt.Errorf("failed to encrypt body: %w", err)
+	}
+
+	encoded := base64.StdEncoding.EncodeToString(eBody)
+
+	if opts.OrgName != "" {
+		err = putOrgSecret(client, pk, host, *opts, encoded)
+	} else {
+		err = putRepoSecret(client, pk, baseRepo, opts.SecretName, encoded)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to set secret: %w", err)
+	}
+
+	if opts.IO.IsStdoutTTY() {
+		cs := opts.IO.ColorScheme()
+		fmt.Fprintf(opts.IO.Out, "%s Set secret %s\n", cs.SuccessIcon(), opts.SecretName)
```

Comment:
I think make senses to add in the repo/organization name where the secret was set to this message.